src/nrniv/CMakeLists.txt

```diff
@@ -322,3 +322,55 @@ endif()
 # ~~~
 install(TARGETS nrniv nocmodl modlunit DESTINATION bin)
 install(TARGETS nrniv_lib DESTINATION ${NRN_INSTALL_SHARE_LIB_DIR})
+if(MINGW)
+  set(mswindir ${PROJECT_SOURCE_DIR}/src/mswin)
+  add_custom_command(OUTPUT nrnicon.res
+    COMMAND echo "id ICON \"${mswindir}/nrniv10.ico\"" > nrnicon.rc
+    COMMAND windres nrnicon.rc -O coff -o nrnicon.res
+    DEPENDS ${mswindir}/nrniv10.ico)
+  set_source_files_properties(
+    ${mswindir}/extra/neuron.cpp
+    ${mswindir}/extra/mos2nrn.cpp
+    ${mswindir}/extra/nrnbinstr.cpp
+    PROPERTIES COMPILE_DEFINITIONS MINGW)
+  add_executable(neuron ${mswindir}/extra/neuron.cpp nrnicon.res)
+  add_executable(mos2nrn ${mswindir}/extra/mos2nrn.cpp)
+  add_executable(nrnbinstr ${mswindir}/extra/nrnbinstr.cpp)
+  install(TARGETS neuron mos2nrn nrnbinstr DESTINATION bin)
+endif()
+
+# =============================================================================
+# Build a Windows setup.exe distribution.
+# =============================================================================
+if(MINGW AND (CMAKE_INSTALL_PREFIX MATCHES "C:/marshalnrn/nrn"))
+  # Presumes previous 'make install' with -DCMAKE_INSTALL_PREFIX="C:/marshalnrn/nrn"
+  # Note: see hard coded set for dos_marshal_dir in ConfigFileSetting.cmake
+  add_custom_target(
+    setup_exe
+    COMMAND bash ${PROJECT_SOURCE_DIR}/mingw_files/nrnmingwenv.sh ${CMAKE_INSTALL_PREFIX} -cmake ${PROJECT_SOURCE_DIR}
+    COMMAND "c:/Program Files (x86)/NSIS/makensisw" ${PROJECT_BINARY_DIR}/src/mswin/nrnsetupmingw.nsi
```

Comment:
Instead of hard coding path, I would suggest to make this as CMake variable (e.g. `MAKENSISW_EXE`) which will be set on command line by user as `cmake .. -D MAKENSISW_EXE =...`

Reply:
With respect to '> nrnicon.rc' I tried exactly that but the trial and error attempts to escape properly were taking too much time. I'll try again now that things are basically working. 
I'll move the lists to cmake/NeuronFileLists.cmake
I'll introduce a MAKENSIS_EXE variable.  Where does one do that? I did it just above the target.